web/src/engine/websites/DingManhua.ts

```diff
@@ -0,0 +1,42 @@
+import { Tags } from '../Tags';
+import icon from './DingManhua.webp';
+import { Chapter, DecoratableMangaScraper, type Manga } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON } from '../platform/FetchProvider';
+
+type APIChapters = {
+    data: {
+        chapters: {
+            id: number,
+            contentId: number,
+            chapterName : string
+        }[]
+    }
+}
+
+@Common.MangaCSS(/^{origin}\/comic\/\d+\.html$/, 'div.info > p')
+@Common.MangasMultiPageCSS('/sort?category=0&tag=0&page={page}', 'div.card-content > a')
+@Common.PagesSinglePageCSS('div#comicContainer img.comic-image')
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('dingmanhua', 'DingManhua', 'https://dingmanhua.com', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Media.Manga, Tags.Language.Chinese, Tags.Source.Aggregator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const mangaId = manga.Identifier.match(/comic\/(\d+)\.html$/)[1];
```

Comment:
Expression can be simplified `const [ mangaID ] = manga.Identifier.match(/\d+/);`